File: backend-gin/internal/bin/test_hf.go
package bin

import (
	"backend/internal/services/instances"
	"context"
	"fmt"
)

func TestHuggingFace() {
	hf := instances.GetHfClient()
	ctx := context.Background()
	vec, err := hf.EmbedText(ctx, "I am trying to debug my code in go")
	if err != nil {
		fmt.Println(err)
	} else {
		fmt.Println(vec)
	}
}

File: backend-axum/src/bin/test_hf.rs
use backend::services::instances::get_hf_client;

#[tokio::main]
async fn main() {
    let hf_client = get_hf_client().await;
    let result = hf_client
        .embed_text("I am trying to debug my code in rust")
        .await;
    match result {
        Ok(vec) => println!("{:?}", vec),
        Err(err) => println!("Something went wrong: {}", err),
    };
}
